modules/plugin-search.php

```diff
@@ -506,5 +508,28 @@ class="jetpack-plugin-search__dismiss"
 		return $links;
 	}
 
+}
 
+/**
+ * Master control that checks if Plugin search hints is active.
+ *
+ * @since 7.1.1
+ *
+ * @return bool True if PSH is active.
+ */
+function jetpack_is_psh_active() {
+	if ( false === ( $status = get_transient( 'jetpack_psh_status' ) ) ) {
+		$response = wp_remote_get( 'https://jetpack.com/psh-status/', array( 'sslverify' => false ) );
+		if ( is_wp_error( $response ) ) {
+			return false;
+		}
+		$body = wp_remote_retrieve_body( $response );
+		if ( empty( $body ) ) {
+			return false;
+		}
+		$json = json_decode( $body );
+		$status = (bool) $json->active;
```

Comment:
For some reason I'm having a problem talking to my sandbox, but in doing so I uncovered that it's possible to make this request and have this not be an object at all. 

Let's add a layer of defense like an `isset` or `empty` check. 

Something that will not throw errors no matter what happens.

Reply:
I figured out why it's not working. The page needs to be created on jetpack.com with that URL. Otherwise the body returns the source of a 404 page :)

Reply:
Yup, I know it returns a 404 because I checked with `wp_remote_retrieve_body` and it was returning a 404 because in the diff we're returning the json and exiting. I'll add a check anyway.